protocol/op-deployer-embedded-artifacts.md

```diff
@@ -0,0 +1,67 @@
+# OP Deployer Embedded Artifacts
+
+## Summary
+
+<!-- Most (if not all) documents should have a summary.
+While the length will likely be proportional to the length of the full document,
+the summary should be as succinct as possible. -->
+
+We propose embedding the contract artifacts for a given contracts version inside the op-deployer binary itself. This
+will reduce the complexity of our deployments, and prevent OP Deployer and the contract artifacts from becoming out
+of sync.
+
+## Problem Statement + Context
+
+Today, OP Deployer lets users provide an "artifacts locator" which tells the tool where to find the contract
+artifacts to use for deployment. The locator can be a `file://` url, an `https://` url, or a `tag://` URL. File and
+HTTPs locators are only used in dev for special cases like unit tests, and therefore are not a concern here. Tagged
+locators, however, are used for production deployments and are a huge area of confusion for users for the following
+reasons:
+
+- The artifacts that a tagged locator points to are created from a different branch than OP Deployer itself. This
+  can lead to a mismatch between OP Deployer's deployment code and the artifacts it is trying to deploy.
+- There is a 1:1 mapping between each OP Deployer version and the tagged versions it supports. `v0.2.0` of OP
+  Deployer, for example, only supports smart contracts from `op-contracts/v2.0.0`. However, the version of the smart
+  contracts can sometimes change by minor or patch versions. This makes it difficult to determine which version of
+  the contracts each version of OP Deployer supports, and can lead to unexpected behavior or broken deployments.
+
+In addition, maintaining the tagged locators themselves is non-trivial amount of work:
+
+- New releases require contract artifacts to be built, pushed to GCP, and added to `standard.go`. This means that
+  every new contract release requires a corresponding code change to OP Deployer.
+- Because the artifacts each locator points to are on GCP, they need to be validated client-side before being used.
+  This adds complexity, overhead, and potential for security vulnerabilities.
```

Comment:
For my own understanding this validation is done using the values in standard.go right?

Reply:
Yep